application/app/helpers/dataverse/datasets_helper.rb

```diff
@@ -16,6 +16,26 @@ def verify_file(file)
     retrictions_service.validate_dataset_file(file)
   end
 
+  def link_to_dataset_prev_page(dataverse_url, persistent_id, page, html_options = {})
+    unless page.first_page?
+      url_options = {}
+      url_options[:dv_port] = params[:dv_port]
```

Comment:
Not sure about using the params object in a helper for the views.

At this point, isn't the dataverse_url fully constructed with the right port and scheme?
Can we get it from dataverse_url?
Or from another URL that is the fully formed from the current request?

Reply:
fixed